Spigot-API-Patches/0057-Add-UnknownCommandEvent.patch

```diff
@@ -0,0 +1,85 @@
+From 3395e2bb0176e4be7bbe798ee9eebf67204f9b24 Mon Sep 17 00:00:00 2001
+From: Sweepyoface <github@sweepy.pw>
+Date: Wed, 14 Jun 2017 11:12:01 -0400
+Subject: [PATCH] Add UnknownCommandEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/command/UnknownCommandEvent.java b/src/main/java/org/bukkit/event/command/UnknownCommandEvent.java
+new file mode 100644
+index 00000000..cdd349c3
+--- /dev/null
++++ b/src/main/java/org/bukkit/event/command/UnknownCommandEvent.java
+@@ -0,0 +1,70 @@
++package org.bukkit.event.command;
++
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.Event;
++
++/**
++ * Thrown when a player executes a command that is not defined
++ */
++public class UnknownCommandEvent extends Event {
++    private static final HandlerList handlers = new HandlerList();
++    private CommandSender sender;
++    private String commandLine;
++    private String message;
++
++    public UnknownCommandEvent(final CommandSender sender, final String commandLine, final String message) {
++        super(false);
++        this.sender = sender;
++        this.commandLine = commandLine;
++        this.message = message;
++    }
++
++    /**
++     * Gets the CommandSender or ConsoleCommandSender
++     * <p>
++     *
++     * @return Sender of the command
++     */
++    public CommandSender getSender() {
++        return sender;
++    }
++
++    /**
++     * Gets the command that was send
++     * <p>
++     *
++     * @return Command sent
++     */
++    public String getCommandLine() {
++        return commandLine;
++    }
++
++    /**
++     * Gets message that will be returned
++     * <p>
++     *
++     * @return Unknown command message
++     */
++    public String getMessage() {
++        return message;
++    }
++
++
++    /**
++     * Sets message that will be returned
++     */
```

Comment:
per other review comments / irc - this isn't an ideal case for Cancellable, as you can't "cancel" the fact an unknown command was issued. A null message is very clear that no message should be sent to the sender.